contracts/PenaltyDistributor.vy

```diff
@@ -0,0 +1,498 @@
+         less than `max_epoch`, the account may claim again.
+    @param _addr Address to claim fees for
+    @return uint256 Amount of fees claimed in the call
+    """
+    assert not self.is_killed
+
+    if block.timestamp >= self.time_cursor:
+        self._checkpoint_total_supply()
+
+    last_token_time: uint256 = self.last_token_time
+
+    if self.can_checkpoint_token and (block.timestamp > last_token_time + TOKEN_CHECKPOINT_DEADLINE):
+        self._checkpoint_token()
+        last_token_time = block.timestamp
+
+    last_token_time = last_token_time / WEEK * WEEK
+
+    amount: uint256 = 0
+    user_epoch: uint256 = 0
+    week_cursor: uint256 = 0
+    max_user_epoch: uint256 = 0
+    is_update: bool = True
+    amount, user_epoch, week_cursor, max_user_epoch, is_update = self._claim(_addr, self.voting_escrow, last_token_time)
+
+    if is_update:
+      self.user_epoch_of[_addr] = user_epoch
+      self.time_cursor_of[_addr] = week_cursor
+      log Claimed(_addr, amount, user_epoch, max_user_epoch)
+
+    if amount != 0:
+        token: address = self.token
+        assert ERC20(token).transfer(_addr, amount)
+        self.token_last_balance -= amount
+
+    return amount
+
+
+@external
+@nonreentrant('lock')
+def claim_many(_receivers: address[20]) -> bool:
+    """
+    @notice Make multiple fee claims in a single call
+    @dev Used to claim for many accounts at once, or to make
+         multiple claims for the same address when that address
+         has significant veCRV history
+    @param _receivers List of addresses to claim for. Claiming
+                      terminates at the first `ZERO_ADDRESS`.
+    @return bool success
+    """
+    assert not self.is_killed
+
+    if block.timestamp >= self.time_cursor:
+        self._checkpoint_total_supply()
+
+    last_token_time: uint256 = self.last_token_time
+
+    if self.can_checkpoint_token and (block.timestamp > last_token_time + TOKEN_CHECKPOINT_DEADLINE):
+        self._checkpoint_token()
+        last_token_time = block.timestamp
+
+    last_token_time = last_token_time / WEEK * WEEK
+    voting_escrow: address = self.voting_escrow
+    token: address = self.token
+    total: uint256 = 0
+
+    amount: uint256 = 0
+    user_epoch: uint256 = 0
+    week_cursor: uint256 = 0
+    max_user_epoch: uint256 = 0
+    is_update: bool = True
+
+    for addr in _receivers:
+        if addr == ZERO_ADDRESS:
+            break
+
+        amount, user_epoch, week_cursor, max_user_epoch, is_update = self._claim(addr, voting_escrow, last_token_time)
+
+        if is_update:
+          self.user_epoch_of[addr] = user_epoch
+          self.time_cursor_of[addr] = week_cursor
+          log Claimed(addr, amount, user_epoch, max_user_epoch)
+
+        if amount != 0:
+            assert ERC20(token).transfer(addr, amount)
+            total += amount
+
+    if total != 0:
+        self.token_last_balance -= total
+
+    return True
+
+@external
+def updateReward(_addr: address):
```

Comment:
notice how I had to add a dummy updateReward() method since we expect same signature in [VotingEscrow.vy](https://github.com/ribbon-finance/ribbonomics/blob/d344d6752c1520f71aaf6600ecd376b6c8776c5e/contracts/VotingEscrow.vy#L379)